### PR TITLE
Add more usage information

### PR DESCRIPTION
**Pull request by annawinkler**
Add more information about how to use DomainModel, including info about the field attributes, extending validations, and how to use flat_errors.

### CODE REVIEW

**Comment by annawinkler** [comment]
Yeah - isn't this a PR? :)

**Comment by annawinkler** [comment]
Would you want to merge this in?

File: README.md [hunk 1]
```diff
@@ -1,6 +1,74 @@
...
+
+
+The name of your field (or attribute) is required. When you specify a field, you get getters and setters automatically:
+
+	red_fish = Fish.new
+	red_fish.color = "red"
+	red_fish.habitat = Habitat.new({
+	  :temperature => 75.0,
+	  :bottom_type => "sand"
+	)
```
Comment (rafer) [inline README.md:42]: EFORGOTTENCURLYBRACE

**Review by rafer — APPROVED** [review]
Thanks very much @annafw!

**Pull request closed without merging**